frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/components/assisted-installer/utils.ts

```diff
@@ -161,12 +161,11 @@ export const onHostsNext = async ({ values, clusterDeployment, agents, agentClus
 }
 
 /** AI-specific version for the CIM-flow's onHostsNext() callback */
-export const onDiscoveryHostsNext = async ({ clusterDeployment, agents, agentClusterInstall }: OnDiscoverHostsNext) => {
-    // TODO(mlibra): So far we do not need "values" of the Formik - options the user will choose from will come later (like CNV or OCS)
-    // So far no need to "release" agents since the user either deletes the agent or keep the list untouched. Reconsider when "disable" gets in place.
-
+export const onDiscoveryHostsNext = async ({ values,clusterDeployment, agents, agentClusterInstall }: OnDiscoverHostsNext) => {
+    const hostIds =  values.selectedHostIds
     const name = clusterDeployment.metadata.name
     const namespace = clusterDeployment.metadata.namespace
+    agents = agents.filter((a) => !hostIds.includes(a.metadata.uid))
 
     await addAgentsToCluster({
         agents,
```

Comment:
Lets not assign back to `agents`. Not a good practice. We can do the filtering inline

```suggestion
    const name = clusterDeployment.metadata.name
    const namespace = clusterDeployment.metadata.namespace

    await addAgentsToCluster({
        agents: agents.filter((a) => !values.selectedHostIds.includes(a.metadata.uid))
```

Reply:
Changed!